File: fruit_shop.py
from sys import stdin,stdout
items = [[int(x)] for x in stdin.readline().split(' ')] ; i = 0
for x in stdin.readline().split(' '): items[i].append(int(x)) ; i+=1
items = sorted(items,key=lambda x: x[0])
print(items)
money = int(stdin.readline().strip())
res = 0
for i in range(3):
    price = items[i][0]
    num = items[i][1]
    n = money // price;
    if n > num:
        n = num    
    res += n
    money = money - (n * price)
stdout.write(str(res)+'\n')

File: FruitShop.c
#include <time.h>
#include <math.h>
#include <stdio.h>
#include <stddef.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>

typedef struct Item{
    int price;
    int num;
}Item;

int cmp(const void *a, const void *b){
    Item i1 = *(Item *)a;
    Item i2 = *(Item *)b;
    return i1.price - i2.price;
}

//int main(int argc, char *argv[]){
//    Item items[3];
//    for(int i = 0; i < 3; i++){
//        scanf("%d", &items[i].price);
//    }
//    for(int i = 0; i < 3; i++){
//        scanf("%d", &items[i].num);
//    }
//    qsort(items, 3, sizeof(Item), cmp);
//    int money;
//    scanf("%d", &money);
//    int res = 0;
//    for(int i = 0; i < 3; i++){
//        int price = items[i].price;
//        int num = items[i].num;
//        int n = money / price;
//        if(n > num){
//            n = num;
//        }
//        res += n;
//        money = money - n * price;
//    }
//    printf("%d\n", res);
//    return 0;
//}
